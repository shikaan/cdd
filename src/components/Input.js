import styled from "styled-components";

import { Size } from "~theme";

const borderWidth = new Size(0.25);

const Input = styled.input(({ theme }) => `
  display: block;
  width: 100%;
  background: transparent;
  height: ${theme.typography.baseFontSize.multiply(2)};
  border: ${borderWidth} solid transparent;
  border-bottom: ${borderWidth} solid ${theme.color.mediumGrey};

  &::placeholder {
    font-size: ${theme.typography.baseFontSize};
    font-family: ${theme.typography.secondaryFont.fontFamily};
    color: ${theme.color.darkGrey}
  }
`);

export default Input;
