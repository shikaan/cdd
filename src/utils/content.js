/**
 * @param {Array<{reference: string}>} microcopy
 * @param {string} key
 * @return {string}
 */
export const getMicrocopy = (microcopy, key) => {
  const element = microcopy?.find(i => i.reference === key) ?? {};

  return element.value ?? "";
};

/**
 * @param {Array<{reference: string}>} sections
 * @param {string} key
 * @return {string}
 */
export const getSection = (sections, key) => {
  return sections?.find(i => i.reference === key) ?? {reference: key};
};