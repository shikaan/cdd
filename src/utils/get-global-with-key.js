export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};
