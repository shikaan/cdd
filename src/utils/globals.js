export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};

export const getGlobal = () => { 
  // eslint-disable-next-line no-restricted-globals
  if (typeof self !== "undefined") { return self }
  if (typeof window !== "undefined") { return window } 
  if (typeof global !== "undefined") { return global } 
  throw new Error("unable to locate global object"); 
}; 
