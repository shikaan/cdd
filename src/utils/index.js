export {isLastIndex} from "./is-last-index";
