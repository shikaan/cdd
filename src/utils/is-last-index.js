export const isLastIndex = (list, index) => {
  if (!Array.isArray(list)) {
    return false;
  }

  return index === list.length - 1;
};
