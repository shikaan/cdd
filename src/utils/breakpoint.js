import {numberBreakpoint} from "~theme/variables";
import {getGlobalWithKey} from "~/utils/get-global-with-key";

export const getViewportWidth = () => getGlobalWithKey("innerWidth") ?? getGlobalWithKey("document").documentElement.clientWidth;

export const isMobile = () => {
  const viewportWidth = getViewportWidth();
  return viewportWidth < numberBreakpoint.sm;
};

export const isTablet = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.sm <= viewportWidth && viewportWidth < numberBreakpoint.md;
};

export const isDesktop = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.md <= viewportWidth;
};