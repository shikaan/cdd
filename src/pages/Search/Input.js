import React, { Component } from "react";
import styled from "styled-components";

import Input from "~components/Input";
import {getMicrocopy} from "~/utils";

const Section = styled.section(({ theme }) => `
  padding: ${theme.templateVariables.horizontalPadding};
`);

const StyledInput = styled(Input)(({ theme }) => `
  font-size: ${theme.typography.baseFontSize.multiply(2.5)};
  height: ${theme.typography.baseFontSize.multiply(3)};
  border: none;
  
  &::placeholder {
    font-size: ${theme.typography.baseFontSize.multiply(2.5)};
    color: ${theme.color.dark400};
  }
`);

const InputWrapper = styled.div(({ theme }) => `
  padding: ${theme.templateVariables.horizontalPadding} 0;
`);

class SearchResults extends Component {
  handleInputChange = (event) => {
    const searchQuery = event.target.value.trim().toLowerCase();

    this.props.setSearchQuery(searchQuery, this.props.performSearch);
  };

  render () {
    const { content, searchQuery } = this.props;

    const placeholder = getMicrocopy(content.microcopy, "search.placeholder");

    return (
      <Section>
        <InputWrapper>
          <StyledInput
            value={searchQuery}
            placeholder={placeholder}
            onChange={this.handleInputChange}/>
        </InputWrapper>
      </Section>
    );
  }
}

export default SearchResults;
