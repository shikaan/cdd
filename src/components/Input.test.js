import React from "react";
import "jest-styled-components";

import Input from "./Input";
import {shallowWithTheme} from "/test/utils";
import themeMock from "~theme";

describe("Input", () => {
  it("renders correctly", () => {
    const wrapper = shallowWithTheme(<Input />, themeMock);

    expect(wrapper).toMatchSnapshot();
  });
});
