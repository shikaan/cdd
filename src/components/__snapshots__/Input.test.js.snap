// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          [Function],
        ],
      },
      "displayName": "styled.input",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "input",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
